packages/nextjs/src/server/utils/getAuth.ts

```diff
@@ -0,0 +1,50 @@
+import { createGetToken } from '@clerk/backend-core';
+import { AuthStatus } from '@clerk/backend-core';
+import { NextRequest } from 'next/server';
+
+import { AuthStatusMapping, SessionsApi } from '../types';
+
+export function createGetAuth(sessionsApi: SessionsApi) {
+  return function (req: NextRequest) {
+    // When the __clerk query param is set, we trust that middleware already ran
+    // Then, we don't have to run any re-verify the JWT here, we
+    // can just strip out the claims manually.
+
+    const clerkParam = req.nextUrl.searchParams.get('__clerk');
+
+    if (!clerkParam) {
+      throw 'You need to use "runClerkMiddleware" on your Next.js middleware file.';
+    }
+
+    // Signed out case
+    if (clerkParam === AuthStatusMapping[AuthStatus.SignedOut]) {
+      return {
+        // TODO what to return here?
+        sessionClaims: null,
+      };
+    }
+
+    // Signed in case
+    if (clerkParam === AuthStatusMapping[AuthStatus.SignedIn]) {
+      // Get the token from cookie or header
+
+      const cookieToken = req.cookies.get('__session');
+      const headerToken = req.headers.get('Authorization')?.replace('Bearer ', '');
+      const token = cookieToken || headerToken || '';
+
+      const authData = JSON.parse(atob(token.split('.')[1]));
+
+      return {
+        ...authData,
+        getToken: createGetToken({
+          headerToken: authData.headerToken,
+          cookieToken: authData.cookieToken,
+          sessionId: authData.sessionId,
+          fetcher: (...args) => sessionsApi.getToken(...args),
+        }),
+        headerToken: headerToken, // TODO return this?
+        cookieToken: cookieToken, // TODO return this?
```

Comment:
I think we shouldn't increase the public API surface unless a customer explicitly asks for these two - we can always expose them going forwards

Reply:
@nikosdouvlis These fields already exist, the question is whether to populate them